src/p5.dance.js

```diff
@@ -287,30 +291,38 @@ module.exports = class DanceParty {
       var msPerBeat = 60 * 1000 / (this.songMetadata_.bpm * (sprite.dance_speed / 2));
       var msPerFrame = msPerBeat / FRAMES;
       while (sprite.sinceLastFrame > msPerFrame) {
+        const totalFrames = sprite.current_move > 12 ? 12 : 24;
+        const currentFrame = sprite.frame;
         sprite.sinceLastFrame -= msPerFrame;
         sprite.looping_frame++;
-        if (sprite.animation.looping) {
-          sprite.animation.changeFrame(sprite.looping_frame % sprite.animation.images.length);
+        if (sprite.isLooping) {
+          sprite.frame = sprite.looping_frame % totalFrames;
         } else {
-          sprite.animation.nextFrame();
+          sprite.frame = (sprite.frame + 1) % totalFrames;
         }
 
         if (sprite.looping_frame % FRAMES === 0) {
-          if (ANIMATIONS[sprite.style][sprite.current_move].mirror) sprite.mirroring *= -1;
-          if (sprite.animation.looping) {
+          if (ANIMATIONS[sprite.style] && ANIMATIONS[sprite.style][sprite.current_move].mirror) {
+            sprite.mirroring *= -1;
+          }
+          if (sprite.isLooping) {
             sprite.mirrorX(sprite.mirroring);
           }
         }
 
-        var currentFrame = sprite.animation.getFrame();
-        if (currentFrame === sprite.animation.getLastFrame() && !sprite.animation.looping) {
-          //changeMoveLR(sprite, sprite.current_move, sprite.mirroring);
-          sprite.changeAnimation("anim" + sprite.current_move);
-          sprite.animation.changeFrame(sprite.looping_frame % sprite.animation.images.length);
+        if (currentFrame === totalFrames && !sprite.isLooping) {
```

Comment:
This wasn't working for me - I think because we set sprite.frame `% totalFrames` above so they'll never be equal.  You could compare to totalFrames - 1 but I think that drops the last frame in a non-looping animation?